### PR TITLE
various fixes

### PR DESCRIPTION
**Pull request by jef-n**
* PEP8 fixes
* cleanup imports and replace PyQt5 with qgis.PyQt
* prefix expression functions with exp_ (fixes layers with leading digits)
* handle empty expressions
* skip NoGeometry layers
* replace message log QgsMessageLog enum with Qgis enum
* report unsupported renderers

### CODE REVIEW

File: qgis2web/leafletScriptStrings.py [hunk 1]
```diff
@@ -246,14 +245,15 @@ def iconLegend(symbol, catr, outputProjectFileName, layerName, catLegend, cnt):
 
 def pointToLayerFunction(safeLayerName, sl):
     try:
-        if isinstance(sl, QgsSvgMarkerSymbolLayerV2):
+        if isinstance(sl, QgsSvgMarkerSymbolLayer):
```
Comment (tomchadwin) [inline qgis2web/leafletScriptStrings.py]: Arg. How did I miss this? Thanks very much.

File: qgis2web/maindialog.py [hunk 2]
```diff
@@ -353,11 +345,10 @@ def populate_layers_and_groups(self, dlg):
...
         for tree_layer in tree_layers:
             layer = tree_layer.layer()
-            if (layer.type() != QgsMapLayer.PluginLayer and
-                    layer.customProperty("ol_layer_type") is None):
+            if (layer.type() != QgsMapLayer.PluginLayer and (layer.type() != QgsMapLayer.VectorLayer or layer.wkbType() != QgsWkbTypes.NoGeometry) and layer.customProperty("ol_layer_type") is None):
                 try:
-                    if layer.type() == QgsMapLayer.VectorLayer:
-                        testDump = layer.renderer().dump()
+                    # if layer.type() == QgsMapLayer.VectorLayer:
+                    #    testDump = layer.renderer().dump()
```
Comment (tomchadwin) [inline qgis2web/maindialog.py]: There was a reason for this, long ago, but I'd have to dig out the commit to see what issue it solved. I'll check blame.

File: qgis2web/resources.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 #
 # WARNING! All changes made in this file will be lost!
 
-from PyQt4 import QtCore
```
Comment (tomchadwin) [inline qgis2web/resources.py]: Arg!

File: qgis2web/leafletLayerScripts.py [hunk 4]
```diff
@@ -183,9 +180,9 @@ def getLabels(layer, safeLayerName, outputProjectFileName, vts, vtLabels):
             fontItalic = font.italic()
             fontBold = font.bold()
             fontColor = text.color().name()
-            fontUnderline = font.underline()
-            xOffset = palyr.xOffset
-            yOffset = palyr.yOffset
+            # fontUnderline = font.underline()
```
Comment (tomchadwin) [inline qgis2web/leafletLayerScripts.py]: Are these implemented in a different way now? I'll find out.

**Comment by tomchadwin** [comment]
This is immense! Thank you so much for all the work. Are you happy for me to merge? I'll either fix PEP8 or ignore the line-length rule.

**Comment by jef-n** [comment]
> This is immense! Thank you so much for all the work. Are you happy for me to merge? I'll either fix PEP8 or ignore the line-length rule.

Yes, fine with me - apparently I unable to break those long lines without triggering a warning - one or the other.  I also didn't find a clear recipe how to do it.

File: qgis2web/leafletLayerScripts.py (hunk 4 above)
Comment (jef-n) [reply qgis2web/leafletLayerScripts.py]: not it was just complaining that these were not actually used anywhere.

File: qgis2web/maindialog.py (hunk 2 above)
Comment (jef-n) [reply qgis2web/maindialog.py]: maybe for debugging purposes - testDump is not actually used anywhere - and failed when layers without a renderer (NoGeometry) still got here.

Comment (tomchadwin) [reply qgis2web/maindialog.py]: I think this was simply to test whether the layer could be queried for its renderer, and to skip the layer if not.

Comment (tomchadwin) [reply qgis2web/maindialog.py]: Yes: https://github.com/tomchadwin/qgis2web/commit/390431c974b6b46fce53598cc0a6fd7fce6f6783#diff-3a7deece77edeafc97a5df780ca99df0R625. You are now testing explicitly for layers without geometry, so this should be fine, unless it was catching other undetected cases where the layer should be skipped.

**Pull request merged**